src/browser/app-state.js

```diff
@@ -20,6 +20,9 @@ var onSetConnectionState = function(currentState, connectionState){
 };
 
 var onAddTeam = function(currentState, team){
+  if (!team.name) {
+    team.name = team.url.replace(/[^:]*:\/\/[^\/]*\/([^/]*)(\/.*|$)/, '$1')
```

Comment:
This isn't the correct place for this if it intends to solve #73.  This file manages state, which is initialized from the loaded configuration.  By the time we get to this file, we need to have already handled the missing `team.name`

This check *should* be inside the `src/load-settings.js` as a 'validateState()` function, but a simple near-term solution would be to put the check within the component itself.